CheckerUI_xforms/CheckerUI/CheckerUI/CheckerUI/Helpers/Order/OrderItemView.cs

```diff
@@ -1,70 +1,79 @@
-﻿using System;
-using System.Collections.Generic;
-using System.ComponentModel;
-using System.Text;
+﻿using System.Collections.Generic;
 using CheckerUI.Helpers.Order;
 using CheckerUI.Models;
 using CheckerUI.ViewModels;
 using Xamarin.Forms;
 using FontAwesome;
-using Prism.Behaviors;
 
+//Here we produce the display of a single order item in its line, the grid consists of three rows (currently)
+//In the first row, the name of the dish and an ID card
+//In the second row, the status of the dish varies depending on the button being pressed
+//And in the third row three buttons are linked to the command respectively
+//An order item knows to update those responsible for a change in its status
+
+
+// To Do:  Status should be changed to Enum
 namespace CheckerUI.Helpers
 {
     public class OrderItemView : BaseViewModel
     {
+        //Properties
         private Grid m_Grid { get; set; }
-        public OrderButtonModel m_Button { get; set; }
-
         private readonly List<Label> m_Labels;
-
+        private readonly OrderItemModel m_orderItem;
         private readonly Button m_DoneButton;
         private readonly Button m_StartButton;
         private readonly Button m_StopButton;
+        private readonly Button m_RestoreButton;
+
+        private bool m_OrderRestored = false;
+        // Commands
         public Command StopCommand { get; }
         public Command StartCommand { get; }
         public Command DoneCommand { get; }
         public Command ReadyCommand { get; }
+        public Command RestoreCommand { get; }
 
-        private readonly OrderItemModel m_orderItem;
-        private string m_OrderStatusText;
+        public OrderItemView()
+        {
 
-        public OrderItemView(string i_Name, OrderButtonModel i_Button, BaseLineViewModel.OrderIDNotifier idNotifier)
+        }
+        public OrderItemView(string i_Name, OrderIDNotifier idNotifier)
         {
-            m_Button = new OrderButtonModel();
-            m_Button = i_Button;
+           
             m_orderItem = new OrderItemModel();
-            m_orderItem = CreateItemView(idNotifier.OrderID, i_Name, idNotifier.Status, 100, "some notes", 1);
+            m_orderItem = CreateItemView( i_Name,  100, "Notes ...", 1, idNotifier);
             CreateGrid();
             ReadyCommand = new Command(() =>
             {
                 if (OrderStatus == 0)
                 {
-                    m_Labels[1].Text = "Available";
-                    idNotifier.Status = 2;
-                    m_OrderStatusText = OrderStatusToString();
+                    OrderStatus = 1;
+                    m_Labels[2].Text = idNotifier.StatusString;
+                    OrderStatusColor = idNotifier.StatusColor;
+                    OrderStatusString = idNotifier.StatusString;
```

Comment:
1) These 4 lines are duplicated in the definitions of the commands below. 
IMO it will be better to do one function to depict this functionality and call upon it like so:
```suggestion
                    OrderStatus = 1;
                    bindThis(OrderStatus);
                    
                    //  private void bindThis(OrderIDNotifier idNotifier){
                    // m_Labels[2].Text = idNotifier.StatusString;
                    // OrderStatusColor = idNotifier.StatusColor;
                    // OrderStatusString = idNotifier.StatusString;
                    //}
                    
```


2) But WHY do we need 3 commands to do this binding? idNotifier is the same for them all, so the binding will just be redone again and again?
Bind once, for general creation of the item, not inside a command definition, and have the commands do what they will. Since this **is** a reference type and not just a struct.

Reply:
we dont need I forgot to clean that up